config/config.go

```diff
@@ -95,7 +95,7 @@ func DefaultConfig() *Config {
 // otherwise use default config
 func MergedConfig(conf *Config) (*Config, error) {
 	merged := DefaultConfig()
-	err := mergo.Merge(merged, conf, mergo.WithOverride)
+	err := copier.CopyWithOption(merged, conf, copier.Option{IgnoreEmpty: true, DeepCopy: true})
```

Comment:
Do we need deep copy here? I saw you didn't use deep copy in ncp-go and nkn-sdk-go but use it here. It seems deep copy doesn't make a difference in this case since all config fields are either basic type of slice.

Reply:
Since the config contains a slice of transport instances and routing types instead of pointers my understanding is that we would need to deep copy those. See [copier](https://github.com/jinzhu/copier?tab=readme-ov-file#complex-data-copying-nested-structures-with-slices) documentation.